lxd/db/cluster/schema.go

```diff
@@ -12,8 +12,23 @@ CREATE TABLE certificates (
     type INTEGER NOT NULL,
     name TEXT NOT NULL,
     certificate TEXT NOT NULL,
+    restricted INTEGER NOT NULL DEFAULT 0,
     UNIQUE (fingerprint)
 );
+CREATE TABLE certificates_projects (
```

Comment:
@stgraber any reason you've not added an `id` field here, there are other tables that don't technically need an `id` column, such as `storage_volumes_config` that do, so I thought there was some convention or reason to have them (I'm not sure about sqlite, but I remember in MySQL innodb that if you don't create an id column one is effectively created anyway, so you may as well have it explicitly created so it can be used).

Reply:
I based it on `images_profiles` which similarly doesn't have an `id` column.

Reply:
OK did a bit of reading and it seems that sqlite uses an internal `rowid` column for all tables by default. If a column is explicitly defined as a primary key integer field then this is used as an alias of the `rowid` column (i.e its not stored twice). https://sqlite.org/lang_createtable.html#rowid

However in this case, the primary key is really a clustered index over `certificate_id` and `project_id` in which case even if you defined it as a primary key, the rowid internal column would still be created.

The only way to avoid this and have a true clustered index over multiple columns would be to define the primary key over those columns and set the `WITHOUT ROWID` optimisation https://sqlite.org/withoutrowid.html 

Which for this case seems overkill, so is probably fine as it is.